packages/vc-verification/src/issuer-verification.ts

```
@@ -0,0 +1,230 @@
+import * as jwt from 'jsonwebtoken';
+import { providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { CredentialResolver, IssuerDefinitionResolver } from '.';
+import {
+  IVerifiableCredential,
+  VerificationResult,
+  OffChainClaim,
+} from './models';
+import { upgradeChainId } from './upgrade-chainid';
+
+export class IssuerVerification {
+  private _provider: providers.Provider;
+  private _resolver: Resolver;
+  private _registrySetting: RegistrySettings;
+  private _issuerDefResolver: IssuerDefinitionResolver;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param registrySetting
+   * @param credentialResolver
+   * @param issuerDefResolver
+   */
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    credentialResolver: CredentialResolver,
+    issuerDefResolver: IssuerDefinitionResolver
+  ) {
+    this._provider = provider;
+    this._registrySetting = registrySetting;
+    this._issuerDefResolver = issuerDefResolver;
+    this._resolver = new Resolver(this._provider, this._registrySetting);
+    this._credentialResolver = credentialResolver;
+  }
+
+  /**
+   * Verifies chain of trust for a holder's credential
+   * @param credential
+   * @param verifyCredentialProofCallback
+   * @returns
+   */
+  async verifyChainOfTrust(
+    credential: IVerifiableCredential,
+    verifyCredentialProofCallback?: (
+      credential: IVerifiableCredential
+    ) => Promise<VerificationResult>
+  ) {
+    if (credential && verifyCredentialProofCallback) {
+      this.verifyChainOfTrustCallback(
+        credential,
+        verifyCredentialProofCallback
+      );
+    } else if (credential) {
+      this.verifyCredentialChainOfTrust(credential);
+    } else {
+      return 'Insufficient information';
```

Why not to throw an error?